src/components/LoginForm.tsx

```diff
@@ -23,79 +23,129 @@ const labelStyle = {
+                    const userData = { firstName, lastName, email, role };
+                    localStorage.setItem('userData', JSON.stringify(userData));
+                    navigate('/');
+                } catch (error) {
+                    console.error('Login failed:', error);
+                    setStatus('failed');
+                    setFieldError('password', 'Invalid email or password. Please try again.');
+                }
+                setSubmitting(false);
+            }}
+        >
+            {(formik) => (
+                <Form onSubmit={formik.handleSubmit}>
+                    <Stack spacing={4} style={{ width: fieldLength }}>
+                        <FormControl
+                            isRequired
+                            isInvalid={!!(formik.errors.email && formik.touched.email)}
+                        >
+                            <FormLabel htmlFor="email" style={labelStyle}>
+                                Email
+                            </FormLabel>
+                            <Field
+                                as={Input}
+                                type="email"
+                                id="email"
+                                name="email"
+                                style={inputStyle}
+                            />
+                            {formik.errors.email && formik.touched.email && (
+                                <FormErrorMessage>{formik.errors.email}</FormErrorMessage>
+                            )}
+                        </FormControl>
+                        <FormControl
+                            isRequired
+                            isInvalid={!!(formik.errors.password && formik.touched.password)}
+                        >
+                            <FormLabel htmlFor="password" style={labelStyle}>
+                                Password
+                            </FormLabel>
+                            <Field
+                                as={Input}
+                                type="password"
+                                id="password"
+                                name="password"
+                                style={inputStyle}
+                            />
+                            {formik.errors.password && formik.touched.password && (
+                                <FormErrorMessage>{formik.errors.password}</FormErrorMessage>
+                            )}
+                        </FormControl>
+                        <Stack
+                            direction="row"
+                            spacing={4}
+                            display="flex"
+                            alignItems="center"
+                            marginTop={4}
+                            marginBottom={4}
+                        >
+                            <Button
+                                type="submit"
+                                style={{
```

Comment:
I would start using theme.ts, if you need help, pls let me know.

Reply:
Thank you so much for your recommendation. I agree that using theme.ts is more beneficial. I just went ahead and used the fastest method. I will certainly follow your advice in the future to tidy up my code.